nsv13/code/__HELPERS/priority_announce.dm

```diff
@@ -17,6 +17,5 @@
 		if(AR.redalert == state)
 			continue
 		AR.redalert = state
-		spawn(0) //We're branching here so that looping through the areas doesnt depend on the lighting loop finishing. Makes me wish for multithreading :(
 		for(var/obj/machinery/light/L in AR)
-			L.update()
\ No newline at end of file
+			L.update_icon()
```

Comment:
Honestly a CHECK_TICK while in this proc might be a good idea

Reply:
Potentially, but this way it's done in one very brief burst of calculation rather than being dragged out